### PR TITLE
Fix: Remove climetlab to be replaced by earthkit

### LINKED ISSUE #7: Replace `climatlab` with `earthkit`
### What maintenance does this project need?

As `earthkit` is replacing `climatlab`, the dependancy needs to be migrated. 

`/anemoi-inference/src/anemoi/inference/runner.py` needs to be changed

### Organisation

ECMWF

### PR DESCRIPTION
**Pull request by HCookie**
- Replaces climetlab with earthkit
- Added to CHANGELOG
- Adjusted `pyproject.toml`

Closes #7

### CODE REVIEW

**Comment by FussyDuck** [comment]
[CLA assistant check](https://bol-claassistant.ecmwf.int:443/ecmwf/anemoi-inference?pullRequest=8) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://bol-claassistant.ecmwf.int:443/ecmwf/anemoi-inference?pullRequest=8) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bol-claassistant.ecmwf.int:443/check/ecmwf/anemoi-inference?pullRequest=8) it.</sub>

File: src/anemoi/inference/runner.py [hunk 1]
```diff
@@ -36,10 +36,10 @@
 
 
 def forcing_and_constants(source, date, param):
-    import climetlab as cml
+    import earthkit
```
Comment (gmertes) [inline src/anemoi/inference/runner.py:39]: Can you change this to the following, it is the style we use accross anemoi:

`import earthkit.data as ekd`

File: src/anemoi/inference/runner.py [hunk 2]
```diff
@@ -226,6 +226,11 @@ def run(
         if start_datetime is None:
             start_datetime = input_fields.order_by(valid_datetime="ascending")[-1].datetime()
 
+        if isinstance(start_datetime, dict):
+            start_datetime = start_datetime[
+                "base_time"
+            ]  # With earthkit, time is a dictionary with 'base_time' and 'valid_time'
```
Comment (gmertes) [inline src/anemoi/inference/runner.py:232]: Can you put the comment on a new line so the linter doesn't split the assignment into multiple lines. You could put it above the assignment or above the if.

**Review by gmertes — CHANGES_REQUESTED** [review]
Nice!

Am I right that this is not backwards compatible with a climetlab input, because of the changes to datetime? 

(it doesn't need to be compatible, just for my own understanding)

**Comment by HCookie** [comment]
@gmertes At the moment, yes it is not backwards compatible. It might be possible to make it so

File: src/anemoi/inference/runner.py (hunk 2 above)
Comment (HCookie) [reply src/anemoi/inference/runner.py:232]: Resolved in upcoming commit

File: src/anemoi/inference/runner.py (hunk 1 above)
Comment (HCookie) [reply src/anemoi/inference/runner.py:39]: Shall do, twas wondering if there was a shorthand

File: src/anemoi/inference/runner.py (hunk 2 above)
Comment (b8raoult) [reply src/anemoi/inference/runner.py]: Can you use:

`start_datetime = x.order_by(valid_datetime="ascending")[-1].metadata('valid_datetime')`

**Review by gmertes — APPROVED** [review]

**Pull request merged**